Rewrites IBM tutorials to match QSCOUT tutorial template

dowusu-antwi opened the pull request:
See `qss` and `css` notebooks in `docs/source/optimizations/ibm`, e.g., `ibm_compile_css.ipynb`:
![image](https://github.com/SupertechLabs/client-superstaq/assets/18367737/5900bdb4-26b5-4cec-9b7e-51a4d786b43c)

bharat-thotakura commented on docs/source/optimizations/ibm/ibmq_compile_css.ipynb:
```diff
@@ -22,21 +22,27 @@
    ]
```
<p>Typo: should be passed as an argument to css.Service()</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/539/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

dowusu-antwi replied:
Fixed.

bharat-thotakura commented on docs/source/optimizations/ibm/ibmq_compile_css.ipynb:
```diff
@@ -22,21 +22,27 @@
    ]
```
<p>This should be the second circuit: </p><pre class="ql-syntax" spellcheck="false">Line #1.    # Visualize the pulse sequence for the second compiled circuit
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/539/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

dowusu-antwi replied:
Fixed.

bharat-thotakura commented on docs/source/optimizations/ibm/ibmq_compile_qss.ipynb:
```diff
@@ -21,9 +21,37 @@
     "</a>"
```
<p>Maybe worth reducing the output here as well to match the css notebook version:</p><pre class="ql-syntax" spellcheck="false">%pip install -q qiskit-superstaq[examples]
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/539/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

dowusu-antwi replied:
Done.

bharat-thotakura commented on docs/source/optimizations/ibm/ibmq_compile_qss.ipynb:
```diff
@@ -21,9 +21,37 @@
     "</a>"
```
<p>Should we just use qss.SuperstaQProvider() to stay close to the css.Service() in the css notebook?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/539/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

dowusu-antwi replied:
Good catch -- done.

bharat-thotakura commented on docs/source/optimizations/ibm/ibmq_compile_qss.ipynb:
```diff
@@ -21,9 +21,37 @@
     "</a>"
```
<pre class="ql-syntax" spellcheck="false">Line #1.    # Compile with ibmq compile
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/539/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

dowusu-antwi replied:
Thanks, fixed.

bharat-thotakura commented on docs/source/optimizations/ibm/ibmq_compile_qss.ipynb:
```diff
@@ -21,9 +21,37 @@
     "</a>"
```
<p>Following comment from the css notebook:</p><pre class="ql-syntax" spellcheck="false">Line #1.    # Visualize the pulse sequence for the second compiled circuit
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/539/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

dowusu-antwi replied:
Fixed.

bharat-thotakura commented:
I think the notebooks are pretty much in a complete state after some typo fixes (and I think the addendum was a helpful touch!)

bharat-thotakura commented:
LGTM after [3d6b4d2](https://github.com/SupertechLabs/client-superstaq/pull/539/commits/3d6b4d2c7b886ab505b1b3904124780dc81eb201). Will leave final approval to @knsmith @stephanielee9 in case they have further comments.

stephanielee9 commented:
Overall, looks good! A couple things I think we need to update:
* It would be nice if the `ibm.rst` file had a brief description - doesn't need to be very lengthy (could be a couple sentences or a small paragraph) or have references but Bharath put together a nice one for the [QSCOUT tutorial](https://superstaq.readthedocs.io/en/latest/optimizations/qscout/qscout.html)
* The Colab and Binder linked images don't work in the build - [this](https://github.com/SupertechLabs/client-superstaq/issues/525) should work: `[alt text](image_url)](link_url)`
* Nit: "Let us start by creating an example qiskit circuit" <-- that sentence in both notebooks. Should be Cirq in the `css` notebook and maybe capitalize or put in in-line code for the `qss` notebook.

dowusu-antwi commented:
> It would be nice if the `ibm.rst` file had a brief description - doesn't need to be very lengthy (could be a couple sentences or a small paragraph) or have references but Bharath put together a nice one for the [QSCOUT tutorial](https://superstaq.readthedocs.io/en/latest/optimizations/qscout/qscout.html)

Done.

> The Colab and Binder linked images don't work in the build - [this](https://github.com/SupertechLabs/client-superstaq/issues/525) should work: `[alt text](image_url)](link_url)`

Fixed.

> Nit: "Let us start by creating an example qiskit circuit" <-- that sentence in both notebooks. Should be Cirq in the `css` notebook and maybe capitalize or put in in-line code for the `qss` notebook.

Fixed (capitalized).

stephanielee9 reviewed (approved):
lgtm!!

bharat-thotakura commented on docs/source/optimizations/ibm/ibm.rst:
```diff
@@ -1,8 +1,19 @@
 SuperstaQ for IBM
```
nit: 
```suggestion
Superstaq for IBM
```
